core/dbt/cli/main.py

```diff
@@ -0,0 +1,386 @@
+
+
+# dbt clean
+@cli.command("clean")
+@click.pass_context
+@p.profile
+@p.profiles_dir
+@p.project_dir
+@p.target
+@p.vars
+def clean(ctx, **kwargs):
+    """Delete all folders in the clean-targets list (usually the dbt_packages and target directories.)"""
+    flags = Flags()
+    click.echo(f"`{inspect.stack()[0][3]}` called\n flags: {flags}")
+
+
+# dbt docs
+@cli.group()
+@click.pass_context
+def docs(ctx, **kwargs):
+    """Generate or serve the documentation website for your project"""
+
+
+# dbt docs generate
+@docs.command("generate")
+@click.pass_context
+@p.compile_docs
+@p.defer
+@p.exclude
+@p.log_path
+@p.models
+@p.profile
+@p.profiles_dir
+@p.project_dir
+@p.selector
+@p.state
+@p.target
+@p.target_path
+@p.threads
+@p.vars
+@p.version_check
+def docs_generate(ctx, **kwargs):
+    """Generate the documentation website for your project"""
+    flags = Flags()
+    click.echo(f"`{inspect.stack()[0][3]}` called\n flags: {flags}")
+
+
+# dbt docs serve
+@docs.command("serve")
+@click.pass_context
+@p.browser
+@p.port
+@p.profile
+@p.profiles_dir
+@p.project_dir
+@p.target
+@p.vars
+def docs_serve(ctx, **kwargs):
+    """Serve the documentation website for your project"""
+    flags = Flags()
+    click.echo(f"`{inspect.stack()[0][3]}` called\n flags: {flags}")
+
+
+# dbt compile
+@cli.command("compile")
+@click.pass_context
+@p.defer
+@p.exclude
+@p.full_refresh
+@p.log_path
+@p.models
+@p.parse_only
+@p.profile
+@p.profiles_dir
+@p.project_dir
+@p.selector
+@p.state
+@p.target
+@p.target_path
+@p.threads
+@p.vars
+@p.version_check
+def compile(ctx, **kwargs):
+    """Generates executable SQL from source, model, test, and analysis files. Compiled SQL files are written to the target/ directory."""
+    flags = Flags()
+    click.echo(f"`{inspect.stack()[0][3]}` called\n flags: {flags}")
+
+
+# dbt debug
+@cli.command("debug")
+@click.pass_context
+@p.config_dir
+@p.profile
+@p.profiles_dir
+@p.project_dir
+@p.target
+@p.vars
+@p.version_check
+def debug(ctx, **kwargs):
+    """Show some helpful information about dbt for debugging. Not to be confused with the --debug option which increases verbosity."""
+    flags = Flags()
+    click.echo(f"`{inspect.stack()[0][3]}` called\n flags: {flags}")
+
+
+# dbt deps
+@cli.command("deps")
+@click.pass_context
+@p.profile
+@p.profiles_dir
+@p.project_dir
+@p.target
+@p.vars
+def deps(ctx, **kwargs):
+    """Pull the most recent version of the dependencies listed in packages.yml"""
+    flags = Flags()
+    click.echo(f"`{inspect.stack()[0][3]}` called\n flags: {flags}")
+
+
+# dbt init
+@cli.command("init")
+@click.pass_context
+@p.profile
+@p.profiles_dir
+@p.project_dir
+@p.skip_profile_setup
+@p.target
+@p.vars
+def init(ctx, **kwargs):
+    """Initialize a new DBT project."""
+    flags = Flags()
+    click.echo(f"`{inspect.stack()[0][3]}` called\n flags: {flags}")
+
+
+# dbt list
+# TODO: Figure out aliasing for ls (or just c/p?)
```

Comment:
https://click.palletsprojects.com/en/8.1.x/advanced/ I am sure you already found this but just throw it here to make it easier to resolve all TODOs

Reply:
I found a better way around aliases (I think).  Check out the cli_runner function at the top.